include/sys/vm_object.h

```diff
@@ -19,6 +20,8 @@ typedef struct vm_object {
   size_t npages;        /* (@) Number of pages */
   vm_pager_t *pager;    /* Pager type and page fault function for object */
   refcnt_t ref_counter; /* (a) How many objects refer to this object? */
+  vm_object_t *backing_object; /* (@) Pointer to backing object */
+  refcnt_t shadow_counter;     /* (a) For how many objects this is a shadow? */
```

Comment:
If I understand the code correctly, `shadow_counter` is the number of shadow objects for which this object is the __backing__ object. In other words, it's the number of objects immediately (i.e. without other objects in between) shadowing this object.

Reply:
Oh right. My bad. I fixed it.